### PR TITLE
Merge GET branch to main branch

### PR DESCRIPTION
**Pull request by acDante**
- Add support for resuming training from a local checkpoint
- Load candidate entities from json file
- Minor changes in file path and arguments

### CODE REVIEW

File: decode_utils.py [hunk 1]
```diff
@@ -177,9 +177,16 @@ def compute_targets(self, input_ids, input_labels_str):
         ):
 
             # First any positve label can be decoded. This set wll shrink over time.
+
```
Comment (louismartin) [inline decode_utils.py]: NIT: Remove comments

File: decode_utils.py [hunk 2]
```diff
@@ -222,9 +229,11 @@ def compute_targets(self, input_ids, input_labels_str):
                     assert len(child_labels) == 1
                     produced_label = list(child_labels)[0]
 
+                    # For debugging
                     if produced_label not in allowed_labels:
```
Comment (louismartin) [inline decode_utils.py:227]: NIT:maybe add  `print("Produced label not in allowed labels")`

File: load_and_eval_s2s.py [hunk 3]
```diff
@@ -198,12 +198,17 @@ def decode_on_label_lattice(batch_id, input_ids):
 
     print("Loss: {:.3f}".format(loc_loss / loc_steps))
 
+    # # Note: Only use naive approach during prediction
```
Comment (louismartin) [inline load_and_eval_s2s.py]: NIT: remove comments

File: load_and_eval_s2s.py [hunk 4]
```diff
@@ -241,22 +246,34 @@ def decode_on_label_lattice(batch_id, input_ids):
...
-print("# of new labels in dev set:", len(dev_label_set.difference(train_label_set)))
-all_labels_set = train_label_set.union(dev_label_set)
+# Read candidate entities from external file
+all_labels_set = set()
+if args.label_set_file:
+    print("Reading entity list from file for creating entity trie: ", args.label_set_file)
+    with open(args.label_set_file, 'r') as fin:
+        all_labels_set = set(json.load(fin))
+
+    print("Finish loading entitiy set.")
```
Comment (louismartin) [inline load_and_eval_s2s.py]:
```suggestion
    print("Finish loading entity set.")
```

File: local_configs.py [hunk 5]
```diff
@@ -6,4 +6,5 @@
 
 # Cluster-specific configs go here. This file is in gitignore so changes won't 
 # impact other checkouts of the code.
-LOCAL_DATA_DIR = "<REPLACE_THIS>"
\ No newline at end of file
+LOCAL_DATA_DIR = "<DATA_PATH>"
+OUTPUT_DIR = "<CHECKPOINT_PATH>"
```
Comment (louismartin) [inline local_configs.py]: NIT: Add newline

File: s2s_model.py [hunk 6]
```diff
@@ -56,7 +55,8 @@ def compute_metrics(preds, golds):
 def make_s2s_model(model_name="facebook/bart-large", from_file=None, device="cuda"):
     tokenizer = AutoTokenizer.from_pretrained(model_name)
     if from_file is not None:
-        model = AutoModelForSeq2SeqLM.from_pretrained(from_file + "_dir").to(device)
+        # model = AutoModelForSeq2SeqLM.from_pretrained(from_file + "_dir").to(device)
```
Comment (louismartin) [inline s2s_model.py]: NIT: Remove comment

File: data_utils.py [hunk 7]
```diff
@@ -32,20 +34,20 @@ def read_data(self):
...
             new_output = []
             tokenized_labels = []
-            for label in line['output']:
+            for label in line[self.output_key]:
                 tokenized = tuple(tokenizer(label).input_ids[:-1])
                 if tokenized not in tokenized_labels:
                     new_output.append(label)
                     tokenized_labels.append(tokenized)
                 else:
-                    print(f"Line {line['uid']} has repeated labels!")
```
Comment (louismartin) [inline data_utils.py:47]: Won't that cause problems with GROOV models/data?

**Comment by louismartin** [comment]
I'm not super familiar with this codebase but overall looks good to me.

**Comment by louismartin** [comment]
(I cannot approve though)

File: data_utils.py [hunk 8]
```diff
@@ -32,20 +34,20 @@ def read_data(self):
...
-            line['output'] = new_output
+                    print(f"Line {line['id']} has repeated labels!")
```
Comment (kashpop) [inline data_utils.py]: will this be an issue for other GROOV datasets where identifier is stored as 'uid' ?

File: decode_utils.py [hunk 9]
```diff
@@ -177,9 +177,16 @@ def compute_targets(self, input_ids, input_labels_str):
         ):
 
             # First any positve label can be decoded. This set wll shrink over time.
+
+            # for l in labels_str.split(self.sep_token):
+            #     if l not in self.raw_label_id_map:
+            #         print(f"{l} not in raw label map")
+            #         print(f"Label string: {labels_str}")
+
```
Comment (kashpop) [inline decode_utils.py]: if not required, please remove the commented code

File: eval.py [hunk 10]
```diff
@@ -189,7 +193,7 @@ def compute(gold_dataset, guess_dataset, ks=None, inv_propensity_scores_dict=Non
     for guess, gold in zip(guess_dataset, gold_dataset):
         try:
             assert (
-                str(gold["uid"]).strip() == str(guess["id"]).strip()
+                str(gold["id"]).strip() == str(guess["id"]).strip()
```
Comment (kashpop) [inline eval.py]: will this break the compatibility with other datasets used in GROOV?

**Review by kashpop — COMMENTED** [review]
Overall, changes look good to me. Have added some comments where I think the compatibility with GROOV dataset might get affected.

**Comment by louismartin** [comment]
Seems my comments didn't go through before

**Comment by fabiopetroni** [comment]
I can't see a REAME with instructions to run this code. @acDante can you please add that?

File: eval.py (hunk 10 above)
Comment (acDante) [reply eval.py]: Good point! I will add support for both identifiers :-).

File: decode_utils.py (hunk 1 above)
Comment (acDante) [reply decode_utils.py]: Which comments?

**Comment by acDante** [comment]
Thanks for sharing the comments! I have committed some new changes with the following main update:
- Fix the problems in comments
- Add scripts for data pre-processing
- Update the readme file

**Review by fabiopetroni — APPROVED** [review]
LGTM - after you fix the links

**Pull request merged**